source/extensions/common/wasm/context.cc

```diff
@@ -1461,10 +1370,12 @@ void Context::onEvent(Network::ConnectionEvent event) {
 void Context::initializeReadFilterCallbacks(Network::ReadFilterCallbacks& callbacks) {
   network_read_filter_callbacks_ = &callbacks;
   network_read_filter_callbacks_->connection().addConnectionCallbacks(*this);
+  setStreamInfo(callbacks.connection().streamInfo());
```

Comment:
We need to clear the StreamInfo when it is no longer available.  Previously the streaminfo was obtained on every call which required it.  This ensured that we would not have stray pointers.  Even of the streaminfo has a safe lifetime with this change it could change in the future.

Reply:
Agree. Which callback should be clearing the stream info pointer? onHeaders/onBody/etc generally share the same stream info, so it helps with the performance to avoid reconstructing the environment.